src/runtime_src/hip/core/memory.h

```diff
@@ -3,15 +3,193 @@
+      : m_size(sz), m_type(memory_type::hip_memory_type_registered), m_hip_flags(flags), m_host_mem(reinterpret_cast<unsigned char *>(host_mem)), m_device(dev), m_bo(nullptr), m_sync_host_mem_required(true)
+    {
+      assert(m_device);
+
+      // user ptr BO is not supported on NPU Linux driver, hence sync between host_mem and internal xrt::bo object is required before and after kernel run
+      // TODO: set m_sync_host_mem_required to true for device that support user ptr BO 
+      init_xrt_bo();
+    }
+
+    ~memory()
+    { 
+      free_mem(); 
+    }
+
+    memory(size_t sz, std::shared_ptr<xrt::core::hip::device>  dev);
+
+    static void lock_pages(void* addr, size_t size);
+
+    void validate();
+    void pre_kernel_run_sync_host_mem();
+    void post_kernel_run_sync_host_mem();
+
+    void copy_from(const xrt::core::hip::memory *src, size_t size, size_t src_offset = 0, size_t offset = 0);
+    void copy_from(const void *host_src, size_t size, size_t src_offset = 0, size_t offset = 0);
+
+    void copy_to(void *host_dst, size_t size, size_t dst_offset = 0, size_t offset = 0) const;
+
+    void set_device(std::shared_ptr<xrt::core::hip::device> device)
+    { 
+      m_device = device; 
+    }
+
+    void*
+    get_host_addr()
+    {
+      return m_host_mem;
+    }
+
+    void* get_device_addr();
+
+    const std::shared_ptr<xrt::bo> get_xrt_bo() const 
+    { 
+      return m_bo; 
+    }
+    
+    std::shared_ptr<xrt::bo> get_xrt_bo() 
+    { 
+      return m_bo; 
+    }
+
+    unsigned int get_hip_flags() const
+    { 
+      return m_hip_flags; 
+    }
+
+    memory_type get_type() const
+    { 
+      return m_type; 
+    }
+
+    memory_group get_group() const
+    { 
+      return m_group; 
+    }
+    
+    void set_group(memory_group group)
+    {
+      m_group = group; 
+    }
+    
+    std::shared_ptr<xrt::core::hip::device>
+    get_device()
+    {
+      return m_device;
+    }
+
+  private:
+    size_t m_size;
+    memory_type m_type;
+    unsigned int m_hip_flags;
+    unsigned char *m_host_mem;
+    std::shared_ptr<xrt::core::hip::device>  m_device;
+    std::shared_ptr<xrt::bo> m_bo;
+    xrt::memory_group m_group;
+    bool m_sync_host_mem_required;
 
+    void free_mem();
 
-class memory
+    void init_xrt_bo();
+  };
+
+class address_range_key
 {
+public:
+    address_range_key() : address(0), size(0) {}
+    address_range_key(uint64_t addr, size_t sz) : address(addr), size(sz) {}
+
+    uint64_t    address;
+    size_t      size;
+};
 
+struct addre_sz_key_compare
+{
+    bool operator() (const address_range_key& lhs, const address_range_key& rhs) const
+    {
+        return ((lhs.address + lhs.size)  < rhs.address);
+    }
 };
-  
+
+typedef std::map<address_range_key, std::shared_ptr<xrt::core::hip::memory>, addre_sz_key_compare> XRT_HIP_ADDR_MAP;
+typedef XRT_HIP_ADDR_MAP::iterator XRT_HIP_ADDR_MAP_ITR;
```

Comment:
Remove.
Do not use `typedef` but `using` since 2011.
Only uppercase for macros.
Since you are in a namespace, just use
```suggestion
using addr_map = std::map<address_range_key, std::shared_ptr<xrt::core::hip::memory>, addre_sz_key_compare>;
using addr_map_iter = addr_map::iterator;
```

Reply:
fixed.